milli/src/search/facet/filter.rs

```diff
@@ -332,15 +341,18 @@ impl<'a> Filter<'a> {
         rtxn: &heed::RoTxn,
         index: &Index,
         filterable_fields: &HashSet<String>,
+        universe: Option<&RoaringBitmap>,
     ) -> Result<RoaringBitmap> {
         match &self.condition {
             FilterCondition::Not(f) => {
+                // TODO improve the documents_ids to also support intersections at deserialize time.
                 let all_ids = index.documents_ids(rtxn)?;
                 let selected = Self::inner_evaluate(
                     &(f.as_ref().clone()).into(),
                     rtxn,
                     index,
                     filterable_fields,
+                    universe,
```

Comment:
I think `all_ids` here is simply the universe or this call to `documents_ids` if no universe has been given.